include/aws/common/byte_buf.h

```diff
@@ -51,6 +51,10 @@ static inline struct aws_byte_cursor aws_byte_cursor_from_buf(const struct aws_b
     return (struct aws_byte_cursor){.ptr = buf->buffer, .len = buf->len};
 }
 
+static inline struct aws_byte_cursor aws_byte_cursor_from_array(const uint8_t *c_str, size_t len) {
+    return (struct aws_byte_cursor){.ptr = (uint8_t *)c_str, .len=len};
```

Comment:
I don't think designated initializers work on msvc (unless I'm missing a flag) https://godbolt.org/g/1a7h2e

(sorry i missed that in earlier reviews)